File: backend/src/main/java/com/studhub/course/Course.java
package com.studhub.course;

import com.studhub.user.User;
import jakarta.persistence.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "courses")
public class Course {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 500)
    private String title;

    @Column(length = 4000)
    private String description;

    @Column(name = "short_name", length = 100)
    private String shortName;

    @Column(length = 200)
    private String category;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private CourseStatus status = CourseStatus.ACTIVE;

    @Column(nullable = false)
    private boolean visible = true;

    @Column(name = "enrollment_open", nullable = false)
    private boolean enrollmentOpen = true;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "course", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("enrolledAt DESC")
    private List<CourseEnrollment> enrollments = new ArrayList<>();

    @PrePersist
    void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public String getShortName() { return shortName; }
    public void setShortName(String shortName) { this.shortName = shortName; }

    public String getCategory() { return category; }
    public void setCategory(String category) { this.category = category; }

    public CourseStatus getStatus() { return status; }
    public void setStatus(CourseStatus status) { this.status = status; }

    public boolean isVisible() { return visible; }
    public void setVisible(boolean visible) { this.visible = visible; }

    public boolean isEnrollmentOpen() { return enrollmentOpen; }
    public void setEnrollmentOpen(boolean enrollmentOpen) { this.enrollmentOpen = enrollmentOpen; }

    public User getOwner() { return owner; }
    public void setOwner(User owner) { this.owner = owner; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public LocalDateTime getUpdatedAt() { return updatedAt; }

    public List<CourseEnrollment> getEnrollments() { return enrollments; }
    public void setEnrollments(List<CourseEnrollment> enrollments) { this.enrollments = enrollments; }
}

File: frontend/src/types/course.ts
export type CourseStatus = 'ACTIVE' | 'ARCHIVED' | 'DELETED';

export interface CourseEnrollment {
  userId: number;
  userFullName: string;
  userEmail: string;
  userLogin: string;
  courseRole: 'STUDENT' | 'TEACHER' | 'OWNER';
  enrolledAt: string;
}

export interface Assignment {
  id: number;
  courseId: number;
  title: string;
  description?: string;
  type: 'ASSIGNMENT' | 'QUIZ';
  status: 'ACTIVE' | 'ARCHIVED' | 'DELETED';
  maxScore: number;
  dueDate?: string;
  sortOrder: number;
  createdAt: string;
  updatedAt: string;
}

export interface Course {
  id: number;
  title: string;
  description?: string;
  shortName?: string;
  category?: string;
  status: CourseStatus;
  visible: boolean;
  enrollmentOpen: boolean;
  ownerId: number;
  ownerName: string;
  createdAt: string;
  updatedAt?: string;
  enrollments?: CourseEnrollment[];
  assignments?: Assignment[];
  myRole?: 'OWNER' | 'TEACHER' | 'STUDENT' | null;
}

export interface CreateCourseRequest {
  title: string;
  description?: string;
  shortName?: string;
  category?: string;
  visible?: boolean;
  enrollmentOpen?: boolean;
}

export interface UpdateCourseRequest {
  title?: string;
  description?: string;
  shortName?: string;
  category?: string;
  visible?: boolean;
  enrollmentOpen?: boolean;
}

export interface AddMemberRequest {
  userId: number;
  courseRole?: 'STUDENT' | 'TEACHER';
}